llvm/lib/ProfileData/SampleProfReader.cpp

```diff
@@ -1112,9 +1114,13 @@ std::error_code SampleProfileReaderBinary::readNameTable() {
   bool UseMD5 = useMD5();
 
   NameTable.clear();
+  if (*Size > NameTable.max_size())
```

Comment:
Encapsulate this into a function that performs size check and reserves vector, and returns an error code if too large.